Refactor actor WebFinger resolution logic

pfefferle opened the pull request:
Moved WebFinger resolution from table classes to Remote_Actors::get_acct for consistency and reuse. Removed redundant get_webfinger method from trait-actor-list-table.php and updated usages in blocked, followers, and following tables to use the new static method.

<!--- Provide a general summary of your changes in the Title above -->

Fixes #2166

## Proposed changes:
- Removed duplicate WebFinger resolution logic from the Actor_List_Table trait
- Added centralized `get_acct` method to Remote_Actors class with caching support
- Updated all table classes to use the new centralized method instead of the trait method

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* See unittests

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [X] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Optimized WebFinger lookups by centralizing and caching account resolution for faster, more consistent handling across lists.

</details>

pfefferle committed:
Refactor actor WebFinger resolution logic

Moved WebFinger resolution from table classes to Remote_Actors::get_acct for consistency and reuse. Removed redundant get_webfinger method from trait-actor-list-table.php and updated usages in blocked, followers, and following tables to use the new static method.

matticbot committed:
Add changelog

pfefferle committed:
Update get_acct to return string on failure

Changed the get_acct method to return an empty string instead of WP_Error when a remote actor profile is not found, updating the docblock accordingly for consistency in return type.

pfefferle committed:
Add unit tests for get_acct method in Remote_Actors

Introduces comprehensive tests for the get_acct method, covering scenarios such as returning cached acct values, handling non-existent posts, successful and failed Webfinger lookups, fallback to guessing, and proper removal of acct: prefix. These tests improve reliability and coverage for remote actor account retrieval logic.

obenland commented on includes/collection/class-remote-actors.php:
```diff
@@ -485,4 +486,41 @@ public static function get_public_key( $key_id ) {
+			$acct  = Webfinger::guess( $actor );
+		}
+
+		$acct = \str_replace( 'acct:', '', $acct );
```
`Sanitize::webfinger()`? Not sure if trimming `@` is necessary here.

pfefferle replied:
consistency ftw!

pfefferle committed:
Sanitize Webfinger account before saving meta

Replaces manual 'acct:' removal with Sanitize::webfinger for improved consistency and security when storing the Webfinger account in post meta.
